### PR TITLE
[Backport 2.16] Fix undeclared dependencies on CTR_DRBG (and add test)

### PR DESCRIPTION
**Pull request by mpg**
This is the 2.16 backport of #3378. Differences from the main PR:

- first commit (test suites): the SSL test suite doesn't have the new stuff testing full handshakes, so nothing to change here
- second commit (programs/fuzz): skipped entirely as there is no `programs/fuzz` here
- third upstream commit (all.sh): apart form context conflicts, used `config.pl` instead of `config.py` which is not in 2.16.

### CODE REVIEW

File: tests/scripts/all.sh [hunk 1]
```diff
@@ -787,6 +787,24 @@ component_test_rsa_no_crt () {
...
 }
 
+component_test_no_ctr_drbg () {
+    msg "build: Full minus CTR_DRBG"
+    scripts/config.pl full
+    scripts/config.pl unset MBEDTLS_CTR_DRBG_C
+    scripts/config.pl unset MBEDTLS_PSA_CRYPTO_C # requires CTR_DRBG
+    scripts/config.pl unset MBEDTLS_PSA_CRYPTO_STORAGE_C # requires PSA Crypto
+    scripts/config.pl unset MBEDTLS_PSA_CRYPTO_SE_C # requires PSA Crypto
+    scripts/config.pl unset MBEDTLS_USE_PSA_CRYPTO # requires PSA Crypto
```
Comment (gilles-peskine-arm) [inline tests/scripts/all.sh:797]: There's no PSA in 2.16. There's a tension between keeping backports aligned and keeping them free of things that don't belong there. When it comes to PSA stuff, I think 2.16 shouldn't have any.

Comment (mpg) [reply tests/scripts/all.sh:797]: Good point. Keeping those was not intentional, I agree they don't belong in 2.16. I'll remove them.

**Comment by mpg** [comment]
I've amended the last commit in order to fix the issue found by Gilles.

**Comment by mpg** [comment]
The only failure in pr-head was an infrastructure issues, and pr-merge fully passed, so this is good enough CI for merging.

**Pull request merged**